tests/test_mpi.py

```diff
@@ -2727,6 +2903,71 @@ def test_adjoint_F_no_omp(self, mode):
         self.run_adjoint_F(3)
 
 
+class TestElastic:
+
+    @pytest.mark.parallel(mode=[(1, 'diag')])
+    def test_elastic_structure(self, mode):
```

Comment:
at first glance, it looks like there's a lot of redundant checks that also appear in the huge test above... but I may be wrong